core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleManagerSuite.scala

```diff
@@ -131,4 +138,33 @@ class SortShuffleManagerSuite extends SparkFunSuite with Matchers {
     )))
   }
 
+  Seq("true", "false").foreach { value =>
+    test(s"SPARK-34541: shuffle can be removed when spark.shuffle.useOldFetchProtocol=$value") {
```

Comment:
Is this possible to add in `ShuffleSuite`. If so, we don't need to consider the `spark.shuffle.useOldFetchProtocol` since `ShuffleOldFetchProtocolSuite` would cover the old protocol case.
If it's hard to move it to ShuffleSuite. I'm also OK with the current code.

Reply:
If we move it to `ShuffleSuite`, we should consider  the case of `spark.shuffle.useOldFetchProtoclol = false` ,  and different configurations are also more intuitive to test together.

If we need to move it to ShuffleSuite, I'm alse ok.